pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -135,16 +135,11 @@ def setup_distributed(self) -> None:
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
-        self.setup_optimizers(trainer)
-        self.setup_precision_plugin()
-        optimizers_to_device(self.optimizers, self.root_device)
 
         if self._layer_sync:
             self.model = self._layer_sync.apply(self.model)
 
         self.configure_ddp()
-        self.barrier()
```

Comment:
Are you sure the barrier can be dropped? It is there for a reason, we should probably keep it.